supervision/detection/core.py

```diff
@@ -592,12 +627,27 @@ def __all_not_none(item_list: List[Any]):
         class_id = np.hstack(class_id) if __all_not_none(class_id) else None
         tracker_id = np.hstack(tracker_id) if __all_not_none(tracker_id) else None
 
+        # Extract all unique keys from the data_list
+        all_keys = set().union(*(d.keys() for d in data_list))
```

Comment:
When we merge `Detections` objects we should make sure that their `data` dictionaries have the same `key`.

Reply:
see it now? i am using `warnings.warn("All data payload dictionaries in Detections object do not have the same keys!")` just incase dictionaries dont have the same key, code doesn't silently execute without letting the user know? I thought, raising valueerror could be little tricky in this case. what do you think ?

Reply:
should we raise value error, or just warning ? or None?